src/server/analytic_services/prometheus_reports/base_prometheus_report.js

```diff
@@ -0,0 +1,44 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const prom_client = require('prom-client');
+const config = require('../../../../config.js');
+
+function is_enabled() {
+    return config.PROMETHEUS_ENABLED;
+}
+
+class BasePrometheusReport {
+    constructor() {
+        const client = this.prom_client;
+        this._registry = new client.Registry();
+    }
+
+    get prom_client() {
+        return prom_client;
+    }
+
+    get registry() {
+        return this._registry;
+    }
+
+    get metric_prefix() {
+        return config.PROMETHEUS_PREFIX;
+    }
+
+    get enabled() {
+        return is_enabled();
```

Comment:
Seems like this is just propagating the same config value to another global function and here also to a report method although there is no "enabled" state per report, which got me a bit confused. Is this here for convenience? or maybe can be removed so that global configs will be accessed through the main singleton instance in prometheus_reporting.js

Reply:
Same for metric_prefix()

Reply:
Regarding: `is_enabled/enabled`:
I want outside code to consume the enabled from `prometheus_reporting.js` as it is the interface for the whole system but I cannot implement it in `prometheus_reporting.js` because I want to consume it in the reports` implementations which will create a cyclic dependency that will fail to resolve (as some of this code run in the parse phase. 

I want a single truth and I wanted it to not be a config var but an abstraction on top of that because we have plans to change it in the near future and we might localize it per a report, on top of the global config.

regarding: `metrics_prefix`: 
It is not the same for all reports, and the idea was to let report override the base class prefix, you can see and example in  the `nodejs` report

Reply:
These are a lot of big words for a flag which is in config...
Why aren't you just using config.PROMETHEUS_ENABLED wherever you need to use it?
We are getting too many new lines of code for a very slight value.

Reply:
Ok, I will use `config.PROMETHEUS_ENABLED` from the outside and in the base. I still want every derived class to be able to override if different logic is needed.

Reply:
done